twilio/rest/api/__init__.py

```diff
@@ -1,63 +1,135 @@
-"""
-    This code was generated by
-    ___ _ _ _ _ _    _ ____    ____ ____ _    ____ ____ _  _ ____ ____ ____ ___ __   __
-    |  | | | | |    | |  | __ |  | |__| | __ | __ |___ |\ | |___ |__/ |__|  | |  | |__/
-    |  |_|_| | |___ | |__|    |__| |  | |    |__] |___ | \| |___ |  \ |  |  | |__| |  \
-
-    Twilio - Api
-    This is the public Twilio REST API.
-
-    NOTE: This class is auto generated by OpenAPI Generator.
-    https://openapi-generator.tech
-    Do not edit the class manually.
-"""
-
-from twilio.base.domain import Domain
-from twilio.rest.api.v2010 import V2010
-
-class Api(Domain):
-    def __init__(self, twilio):
-        """
-        Initialize the Api Domain
-
-        :returns: Domain for Api
-        :rtype: twilio.rest.v2010.Api
-        """
-        super(Api, self).__init__(twilio)
-        self.base_url = 'https://Api.twilio.com'
-        self._V2010 = None
-
-    @property
-    def V2010(self):
-        """
-        :returns: Versions v2010 of Api
-        :rtype: twilio.rest.Api.v2010
-        """
-        if self._V2010 is None:
-            self._V2010 = V2010(self)
-        return self._V2010
-    
+from twilio.rest.api import ApiBase
+from warnings import warn
+
+
+class Api(ApiBase):
 
     @property
+    def account(self):
+        warn('account() is deprecated. Use v2010.account() instead.', DeprecationWarning, stacklevel=2)
+        return self.v2010.account
+
     def accounts(self):
-        """
-        :rtype: twilio.rest.v2010.accounts
-        """
+        warn('accounts() is deprecated. Use v2010.accounts() instead.', DeprecationWarning, stacklevel=2)
         return self.v2010.accounts
-    
 
     @property
-    def account(self):
-        """
-        :rtype: twilio.rest.v2010.account
-        """
-        return self.v2010.account
-    
-
-    def __repr__(self):
-        """
-        Provide a friendly representation
-        :returns: Machine friendly representation
-        :rtype: str
-        """
-        return '<Twilio.Api>'
+    def addresses(self):
+        warn('addresses() is deprecated. Use v2010.addresses() instead.', DeprecationWarning, stacklevel=2)
+        return self.account.addresses
+
+    @property
+    def applications(self):
+        warn('applications() is deprecated. Use v2010.applications() instead.', DeprecationWarning, stacklevel=2)
```

Comment:
Actually, we should be keeping the v2010 shortcuts and not marking them as deprecated.

Reply:
The work to keep v2010 shortcuts will be part of DI-2456